metadata-ingestion/src/datahub/ingestion/source/powerbi/proxy.py

```diff
@@ -0,0 +1,911 @@
+        scan_create_endpoint = scan_create_endpoint.format(
+            POWERBI_ADMIN_BASE_URL=PowerBiAPI.ADMIN_BASE_URL
+        )
+
+        def create_scan_job():
+            """
+            Create scan job on PowerBi for the workspace
+            """
+            request_body = {"workspaces": [workspace_id]}
+
+            res = requests.post(
+                scan_create_endpoint,
+                data=request_body,
+                params={
+                    "datasetExpressions": True,
+                    "datasetSchema": True,
+                    "datasourceDetails": True,
+                    "getArtifactUsers": True,
+                    "lineage": True,
+                },
+                headers={Constant.Authorization: self.get_access_token()},
+            )
+
+            if res.status_code not in (200, 202):
+                message = f"API({scan_create_endpoint}) return error code {res.status_code} for workspace id({workspace_id})"
+
+                LOGGER.warning(message)
+
+                raise ConnectionError(message)
+            # Return Id of Scan created for the given workspace
+            id = res.json()["id"]
+            LOGGER.info("Scan id({})".format(id))
+            return id
+
+        def wait_for_scan_to_complete(scan_id: str, timeout: int) -> Any:
+            """
+            Poll the PowerBi service for workspace scan to complete
+            """
+            minimum_sleep = 3
+            if timeout < minimum_sleep:
+                LOGGER.info(
+                    f"Setting timeout to minimum_sleep time {minimum_sleep} seconds"
+                )
+                timeout = minimum_sleep
+
+            max_trial = timeout // minimum_sleep
+            LOGGER.info(f"Max trial {max_trial}")
+            scan_get_endpoint = PowerBiAPI.API_ENDPOINTS[Constant.SCAN_GET]
+            scan_get_endpoint = scan_get_endpoint.format(
+                POWERBI_ADMIN_BASE_URL=PowerBiAPI.ADMIN_BASE_URL, SCAN_ID=scan_id
+            )
+
+            LOGGER.info(f"Hitting URL={scan_get_endpoint}")
+
+            trail = 1
+            while True:
+                LOGGER.info(f"Trial = {trail}")
+                res = requests.get(
+                    scan_get_endpoint,
+                    headers={Constant.Authorization: self.get_access_token()},
+                )
+                if res.status_code != 200:
+                    message = f"API({scan_get_endpoint}) return error code {res.status_code} for scan id({scan_id})"
+
+                    LOGGER.warning(message)
+
+                    raise ConnectionError(message)
+
+                if res.json()["status"].upper() == "Succeeded".upper():
+                    LOGGER.info(f"Scan result is available for scan id({scan_id})")
+                    return True
+
+                if trail == max_trial:
+                    break
+                LOGGER.info(f"Sleeping for {minimum_sleep} seconds")
```

Comment:
Do we usually use this LOGGER convention? I don't recall seeing this.

Reply:
Yes, we do have this in oracle.py and snowflake_v2.py. The difference is I used the all-uppercase LOGGER instead of logger.

Reply:
why?

Reply:
No specific reason. I named it in All caps as per python global constant should be in capital letters. I converted it to all lowercase